Packages/Tracking/OpenXR/API Layer/HandTrackingApiLayerFeature.cs

```diff
@@ -0,0 +1,116 @@
+using JetBrains.Annotations;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine.XR.OpenXR;
+using UnityEngine.XR.OpenXR.Features;
+
+#if UNITY_EDITOR
+using System;
+using UnityEditor;
+using UnityEditor.XR.OpenXR.Features;
+using UnityEditor.Build.Reporting;
+#endif
+
+namespace Ultraleap.Tracking.OpenXR.ApiLayer
+{
+    /// <summary>
+    /// Embeddable Ultraleap OpenXR API Layer.
+    /// </summary>
+#if UNITY_EDITOR
+    [OpenXRFeature(FeatureId = FeatureId,
+        Version = "1.0.0",
+        UiName = "Ultraleap Hand Tracking API Layer",
+        Company = "Ultraleap",
+        Desc = "Embeddable API layer for Ultraleap Hand Tracking",
+        Category = FeatureCategory.Feature,
+        Required = false,
+        OpenxrExtensionStrings = "",
+        BuildTargetGroups = new[] { BuildTargetGroup.Android }
+    )]
+#endif
+    public class HandTrackingApiLayerFeature : OpenXRFeature
+    {
+        [PublicAPI] public const string FeatureId = "com.ultraleap.tracking.openxr.feature.handtracking.api_layer";
+
+        protected override void GetValidationChecks(List<ValidationRule> rules, BuildTargetGroup targetGroup)
+        {
+            // Attempt to check the version of OpenXR for at least 1.0.25 as a proxy for what loader version is
+            // included.
+            rules.Add(new ValidationRule(this)
+            {
+                message = "The OpenXR loader must be at least version 1.0.25 to support embedding the Ultraleap API layer",
+                error = false,
+                checkPredicate = () =>
+                {
+                    var versionParts = Array.ConvertAll(OpenXRRuntime.apiVersion.Split("."), int.Parse);
+                    return versionParts[0] >= 1 && versionParts[1] >= 0 && versionParts[2] >= 25;
```

Comment:
I think this version checking might be insufficient, if I had 2.5.2 it would fail because versionParts[2] would be too small?

it would need something like this:

```
if (versionParts[0] > 1)
{
    return true;
}
else if (versionParts[0] == 1)
{
    if(versionParts[1] > 0)
    {
        return true;
    }
    else if (versionParts[1] == 0 && versionParts[2] >= 25)
    {
        return true;
    }
}
return false;
```

Reply:
Yeah you're right, I'll fix that. I wish unity had a version string comparison function...

Reply:
We could also convert to System.Version and then do a [.CompareTo](https://learn.microsoft.com/en-us/dotnet/api/system.version.compareto?view=net-7.0)?

Reply:
I forgot about `System.Version` I've written a much more succinct version using that.